cmd/local_client.go

```diff
@@ -0,0 +1,174 @@
+	clipkg "github.com/urfave/cli"
+	"go.uber.org/zap/zapcore"
+)
+
+// RunNode starts the Chainlink core.
+func (cli *Client) RunNode(c *clipkg.Context) error {
+	config := updateConfig(cli.Config, c.Bool("debug"))
+	logger.SetLogger(config.CreateProductionLogger())
+	logger.Infow("Starting Chainlink Node " + strpkg.Version + " at commit " + strpkg.Sha)
+
+	err := InitEnclave()
+	if err != nil {
+		return cli.errorOut(fmt.Errorf("error initializing SGX enclave: %+v", err))
+	}
+
+	app := cli.AppFactory.NewApplication(config)
+	store := app.GetStore()
+	pwd, err := passwordFromFile(c.String("password"))
+	if err != nil {
+		return cli.errorOut(fmt.Errorf("error starting app: %+v", err))
+	}
+	err = cli.KeyStoreAuthenticator.Authenticate(store, pwd)
+	if err != nil {
+		return cli.errorOut(fmt.Errorf("error starting app: %+v", err))
+	}
+
+	var user models.User
+	if _, err = NewFileAPIInitializer(c.String("api")).Initialize(store); err != nil && err != errNoCredentialFile {
+		return cli.errorOut(fmt.Errorf("error starting app: %+v", err))
+	}
+	if user, err = cli.FallbackAPIInitializer.Initialize(store); err != nil {
+		return cli.errorOut(fmt.Errorf("error starting app: %+v", err))
+	}
+	logger.Info("API exposed for user ", user.Email)
+	if err := app.Start(); err != nil {
+		return cli.errorOut(fmt.Errorf("error starting app: %+v", err))
+	}
+	defer app.Stop()
+	logNodeBalance(store)
+	logConfigVariables(config)
+	logIfNonceOutOfSync(store)
+
+	return cli.errorOut(cli.Runner.Run(app))
+}
+
+func passwordFromFile(pwdFile string) (string, error) {
+	if len(pwdFile) == 0 {
+		return "", nil
+	}
+	dat, err := ioutil.ReadFile(pwdFile)
+	return strings.TrimSpace(string(dat)), err
+}
+
+func logIfNonceOutOfSync(store *strpkg.Store) {
+	account := store.TxManager.GetActiveAccount()
+	lastNonce, err := store.GetLastNonce(account.Address)
+	if err != nil {
+		logger.Error("database error when checking nonce: ", err)
+		return
+	}
+
+	if localNonceIsNotCurrent(lastNonce, account.GetNonce()) {
+		logger.Warn("The account is being used by another wallet and is not safe to use with chainlink")
+	}
+}
+
+func localNonceIsNotCurrent(lastNonce, nonce uint64) bool {
+	if lastNonce+1 < nonce {
+		return true
+	}
+
+	return false
+}
+
+func updateConfig(config strpkg.Config, debug bool) strpkg.Config {
+	if debug {
+		config.LogLevel = strpkg.LogLevel{Level: zapcore.DebugLevel}
+	}
+	return config
+}
+
+func logNodeBalance(store *strpkg.Store) {
+	balance, err := presenters.ShowEthBalance(store)
+	logger.WarnIf(err)
+	logger.Infow(balance)
+	balance, err = presenters.ShowLinkBalance(store)
+	logger.WarnIf(err)
+	logger.Infow(balance)
+}
+
+func logConfigVariables(config strpkg.Config) {
+	wlc := presenters.NewConfigWhitelist(config)
+	logger.Debug("Environment variables\n", wlc)
+}
+
+// DeleteUser is run locally to remove the User row from the node's database.
+func (cli *Client) DeleteUser(c *clipkg.Context) error {
```

Comment:
Seems worth adding an extra confirmation prompt, warning that all sessions will be killed if you continue.

Reply:
👍 hopefully we can save it for another story/pr.